test/verify/check-metrics

```diff
@@ -1353,18 +1348,16 @@ class TestMetricsPackages(packagelib.PackageCase):
         m.execute('until [ $(systemctl is-enabled pmlogger) = enabled ]; do sleep 1; done')
         # also needs to wait for activating → active
         m.execute('until [ $(systemctl is-active pmlogger) = active ]; do sleep 1; done')
-        # triggers "needs logout"
-        b.click("button:contains('Log out')")
-        b.leave_page()
-        b.click("button:contains('Reconnect')")
-        b.set_val("#login-user-input", "admin")
-        b.set_val("#login-password-input", "foobar")
-        b.click('#login-button')
-        b.enter_page("/metrics")
-        # this is just a fake python3-pcp package
         b.wait_in_text(".pf-v5-c-empty-state", "Metrics history could not be loaded")
         b.wait_in_text(".pf-v5-c-empty-state", "pmlogger.service is failing to collect data")
 
+        # HACK: https://github.com/cockpit-project/cockpit/issues/21093
+        # Re-login as we can't wait till handleSave fully completes as the
+        # pcpconfig dialog is closed and we run code which we can't wait on
+        # which closes the dialog (in enabling firewall support).
```

Comment:
This is really hard to read/understand -- I've read it three times and have no idea what's going on. The description in #21093 is fine, but it's not clear to me at all how not seeing a firewalld error is related to logging out and back in?

In that issue, the dialog is prematurely closed, allegedly the firewalld configuration happens in the background. So the only thing that this would do is to introduce some delay. Why can't this poll `firewall-cmd` instead?

Reply:
Ok let me try to rephrase this:

> This is really hard to read/understand -- I've read it three times and have no idea what's going on. The description in https://github.com/cockpit-project/cockpit/issues/21093 is fine, but it's not clear to me at all how not seeing a firewalld error is related to logging out and back in?

Firewalld is a red herring, what the test did before enabling pmproxy was install `pcp` support which also ends up in the `firewalld` related code which closes the Dialog again. As this happens when the install dialog **and** the pcp config dialog is closed we can't assert or wait on this. (Nor is a firewall rule enabled as we did not select pmproxy)

So we open the pcpconfig dialog again to install redis and `Dialogs.close()` is called making the test always fail in CI (not locally as we are slow enough)